pricegraph/src/lib.rs

```diff
@@ -224,6 +225,42 @@ mod tests {
         assert!(orderbook.bids.is_empty());
     }
 
+    #[test]
+    fn transitive_orderbook_simple() {
+        let user0 = UserId::from_low_u64_le(0);
+        let base: u128 = 1_000_000_000_000;
+        let pricegraph = Pricegraph::new(vec![Element {
+            user: user0,
+            balance: U256::from(2) * U256::from(base),
+            pair: TokenPair { buy: 0, sell: 1 },
+            valid: Validity { from: 0, to: 1 },
+            price: Price {
+                numerator: 2 * base,
+                denominator: base,
+            },
+            remaining_sell_amount: base,
+            id: 0,
+        }]);
+        let orderbook = pricegraph.transitive_orderbook(0, 1, None);
+        assert_eq!(
+            orderbook.asks,
+            vec![TransitiveOrder {
+                buy: 2.0 * base as f64,
+                sell: base as f64,
+            }]
+        );
+        assert_eq!(orderbook.bids, vec![]);
```

Comment:
I think this test is backwards (at this point, I'm not really sure though)... from what I understand ask orders are buying quote token and selling base token, and the transitive orderbook was created with `base = 0` and `quote =1`, here the test asserts that the order buying base for quote is an ask.

Reply:
Maybe @fleupold can confirm though, because I've gotten this backwards so many times I don't trust myself anymore :stuck_out_tongue: